lib/sendgrid/template_mailer.rb

```diff
@@ -0,0 +1,33 @@
+require_relative './recipient'
+require_relative './template'
+require_relative './mail'
+
+module SendGrid
+  class InvalidClient < StandardError; end
+  class InvalidTemplate < StandardError; end
+  class InvalidRecipients < StandardError; end
+
+  class TemplateMailer
+
+    def initialize(client, template, recipients = [])
```

Comment:
Could you add a detailed comment here about what this does and how to use it? This looks like the code that we'll be using 90% of the time when we're interacting with the SendGrid gem. I think it would be useful if you copied the example code in the PR comment here.